### DISCUSSION TITLE
Trying to understand Ory Kratos as a newbie.

### DISCUSSION THREAD

**Discussion by Abdullah1428** [ory/kratos#2665] [Q&A] [answered]
Hi, I am new to ory kratos and having some hard time to get the full picture from the documentation for my use case.

I need to add ory kratos into my backend NodeJS (NESTJS) application and React on the frontend for authentication and user management. I saw the sdk version of ory kratos client and looked at the example also given for ory kratos but have a very simple question for now in this initial phase. 

My question is if I want to add ory kratos into my application, is installing the sdk on the backend enough to get this working?  Like do I need to run the kratos using docker also? or just installing the sdk on the backend and then calling the apis to authenticate users on the ory cloud would work without running anything on docker. Like something that is shown in this video also 
[example](https://www.youtube.com/watch?v=umSRQBVdPjM&t=325s)

My question might be too simple as I might be missing some things and I am still reading the documentation and examples to get a more full picture so thanks for the help in advance.

**Comment by vinckr** [comment, marked as answer]
Hello @Abdullah1428 

> My question is if I want to add ory kratos into my application, is installing the sdk on the backend enough to get this working? Like do I need to run the kratos using docker also? 

The SDK enables your application to talk to Ory Kratos without calling the APIs directly - so yes you also need to run the Ory Kratos service - in Docker,k8s or bare metal install.

> Like something that is shown in this video also [example](https://www.youtube.com/watch?v=umSRQBVdPjM&t=325s)

In this video they use the managed Ory Cloud service - basically Ory runs Ory Kratos/Hydra/Keto for you, manage deploys/upgrades/networking etc. and you can just consume the APIs.

If you are new to Ory I would suggest to [create a (free) Developer account in Ory Cloud](https://console.ory.sh/registration) and then you can already use the SDK like you proposed without running anything in Docker. If you decide later on that self-hosting is crucial for you, you can still migrate easily. For a good intro to Ory Kratos on Docker I suggest to try out [the quickstart](https://www.ory.sh/docs/kratos/quickstart).

**Reply by Abdullah1428** [reply]
Thank you for the detail answer @vinckr, this makes now more sense. I will look into the quickstart and read the documentation in detail. 

Regarding this
"In this video they use the managed Ory Cloud service - basically Ory runs Ory Kratos/Hydra/Keto for you, manage deploys/upgrades/networking etc. and you can just consume the APIs."

If I want to do something like this, how can this be achieved, by creating a project in the ory cloud and then connecting to it through the backend application?

**Reply by vinckr** [reply]
> If I want to do something like this, how can this be achieved, by creating a project in the ory cloud and then connecting to it through the backend application?

Yes exactly! We have a number of starting guides for different frameworks/languages: https://www.ory.sh/docs/examples
Let me know if you run into trouble - also feel free to reach out in the [chat](https://slack.ory.sh/).

**Comment by Abdullah1428** [comment]
@vinckr Hi again I have another question,

So I am trying the Http API for ory-kratos in a Node-JS backend app and I have setup Ory Cloud Project. I am trying the end point for [Login Flow](https://www.ory.sh/docs/kratos/reference/api#operation/initializeSelfServiceLoginFlowWithoutBrowser) but it is mentioned in the explanation that 

```
You MUST NOT use this endpoint in client-side (Single Page Apps, ReactJS, AngularJS) 
nor server-side (Java Server Pages, NodeJS, PHP, Golang, ...) browser applications. 
Using this endpoint in these applications will make you vulnerable to a variety of 
CSRF attacks, including CSRF login attacks.
```

So how should I do the login flow and login route in my Node-JS backend app, by using the SDK or any other method?

Super thanks for being helpful.

**Reply by vinckr** [reply]
Hello @Abdullah1428 

The endpoint you linked is for clients that do not have a browser present - so for example mobile or native clients (iOs, Android, etc.) 

If you implement Ory in an application that runs in a browser, you have to [use this endpoint](https://www.ory.sh/docs/kratos/reference/api#operation/initializeSelfServiceLoginFlowForBrowsers).

See also this document: https://www.ory.sh/docs/kratos/self-service/flows/user-login#operation/getVersion
You have three versions of the flow, one for browser clients, one for browser clients with AJAX and one for clients without browsers.

Feel free to open another discussion on that topic specifically if something remains unclear still!
